### PR TITLE
Azure BYOVNET: Support using existing resource groups in deploying custom resources

### PR DESCRIPTION
**Pull request by paulzhang97**
Created a TF var `create_resource_groups`. If set to `false`, the TF will use existing resource groups instead of creating them.
The default value of the TF var is `true`.

#### Sample var file using existing resource group.
[byovnet.use-existing-rg.auto.tfvars.json](https://github.com/user-attachments/files/18014778/byovnet.use-existing-rg.auto.tfvars.json)

### CODE REVIEW

**Comment by david-yu** [comment]
That approach seems good to me, I wonder if this is also applicable across AWS or GCP.

**Comment by paulzhang97** [comment]
> That approach seems good to me, I wonder if this is also applicable across AWS or GCP.

GCP and AWS don't have the concept of resource group.

File: customer-managed/azure/terraform/resource_groups.tf [hunk 1]
```diff
@@ -16,10 +16,15 @@ locals {
...
+  for_each = var.create_resource_groups ? toset(local.resource_group_names) : []
 
   name     = each.value
   location = var.region
 
   tags = var.tags
 }
+
+data "azurerm_resource_group" "all" {
+  for_each = var.create_resource_groups ? [] : toset(local.resource_group_names)
```
Comment (ncole) [inline customer-managed/azure/terraform/resource_groups.tf:28]: Is it possible that one of the names does not exist?

**Review by ncole — APPROVED** [review]

File: customer-managed/azure/terraform/resource_groups.tf (hunk 1 above)
Comment (paulzhang97) [reply customer-managed/azure/terraform/resource_groups.tf:28]: If user chooses to provide RGs by setting `create_resource_groups` to `false`, the RGS must exist. If does not exist, the data source will fail.

**Pull request merged**